mat64/list_test.go

```diff
@@ -68,6 +68,15 @@ func sameAnswerFloat(a, b interface{}) bool {
 	return a.(float64) == b.(float64)
 }
 
+// sameAnswerFloat returns whether the two inputs are both NaN or within tol
+// of each other.
+func sameAnswerFloatApprox(a, b interface{}) bool {
+	if math.IsNaN(a.(float64)) {
+		return math.IsNaN(b.(float64))
```

Comment:
I'm not sure how important it is but in case of NaN this approximate equality is not commutative.